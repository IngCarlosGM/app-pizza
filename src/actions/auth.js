import getUsers from "../helpers/getUsers";
import { types } from "../types/types";
import { finishLoading, startLoading } from './ui';

export const startLoginEmailPassword = ( email, password ) => {
    return ( dispatch ) => {
        dispatch(startLoading);

        getUsers().then(( resp ) => {

            let user = (resp.data.response.users).find( user => user.email === email);
            
            if( user ) {
                dispatch(login( user.name ));
            }

        }).catch( err => {
            console.log(err);
        });

        dispatch(finishLoading);
    }
}

export const login = ( name ) => ({
    type: types.login,
    payload: {
        name
    }
});

export const startLogout = () => {
    return async(dispatch) => {
        // await firebase.auth().signOut();
        // dispatch( logout() );
        // dispatch( noteLogout() );
    }
}