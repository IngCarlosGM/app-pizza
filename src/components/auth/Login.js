import React from 'react'

import { useDispatch, useSelector } from 'react-redux'
import { startLoginEmailPassword } from '../../actions/auth';
import { useForm } from '../../hooks/useForm';

export const Login = ( { history } ) => {

    const dispatch = useDispatch();
    const { loading } = useSelector( state => state.ui );

    const [ formValues, handleInputChange ] = useForm({
        usuario: '',
        password: ''
    });

    const { usuario, password } = formValues;

    const handleLogin = (e) => {
        e.preventDefault();
        const lastPath = localStorage.getItem('lastPath') || '/';

        dispatch( startLoginEmailPassword( usuario, password ) );
        history.replace( lastPath );
    }

    return (
        <div className="welcome mt-primary mb-primary">
            <img 
                id="logo-aside-best-pizza"
                className="welcome__size-logo"
                src={`../assets/Login-Best-Pizza.png`} 
                alt="Logo de Best pizza"
            />

            <h1 className="mt-secondary welcome__title">Bienvenido</h1>
            <h3 className="mt-third welcome__text">A las mejores pizzas del país</h3>

            <div className="container-inputs mt-secondary">
                
                <input 
                    type="text"
                    placeholder="Usuario"
                    name="usuario"
                    className="input input--block input--user"
                    autoComplete="off"
                    value={ usuario }
                    onChange={ handleInputChange }
                />

                <input 
                    type="password"
                    placeholder="Contraseña"
                    name="password"
                    className="input--block input--password mt-input"
                    value={ password }
                    onChange={ handleInputChange }
                />

            </div>

            <a href="#" className="mt-third pointer">¿Olvidaste tu contraseña?</a>

            <button
                className="btn__primary btn-block mt-secondary"
                onClick={(e) => { handleLogin(e) }}
                disabled= { loading }
            >Iniciar sesión</button>
        </div>
    )
}
