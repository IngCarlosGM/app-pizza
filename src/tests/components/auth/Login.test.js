import { mount } from "enzyme";
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { Login } from '../../../components/auth/Login';
 
const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
    auth: {
        name: 'Pepito Perez',
        logged: true
    },
    ui: {
        loading: false,
        msgError: null
    },
    dashboard: {
        
    }
}

let store = mockStore(initialState);

describe('Testing component Login.js', () => {

    const history = {
        replace: jest.fn()
    }
    
    const wrapper = mount(
        <Provider store= { store }>
            <Login history= { history } />
        </Provider>
    );

    test('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should have a image', () => {
        expect(wrapper.find('img').length).toBe(1);
    });

    test('should have a Login button', () => {
        expect(wrapper.find('button').length).toBe(1);
    });

    test('should to call replace function one times', () => {
        wrapper.find('button').simulate('click');
        expect(history.replace).toBeCalledTimes(1);
    });

    test('should have a forget your password', () => {
        expect(wrapper.find('a').length).toBe(1);
    });
    
});
