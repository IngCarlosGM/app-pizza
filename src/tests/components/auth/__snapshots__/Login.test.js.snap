// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component Login.js should render correctly 1`] = `
<div
  className="welcome mt-primary mb-primary"
>
  <img
    alt="Logo de Best pizza"
    className="welcome__size-logo"
    id="logo-aside-best-pizza"
    src="../assets/Login-Best-Pizza.png"
  />
  <h1
    className="mt-secondary welcome__title"
  >
    Bienvenido
  </h1>
  <h3
    className="mt-third welcome__text"
  >
    A las mejores pizzas del país
  </h3>
  <div
    className="container-inputs mt-secondary"
  >
    <input
      autoComplete="off"
      className="input input--block input--user"
      name="usuario"
      onChange={[Function]}
      placeholder="Usuario"
      type="text"
      value=""
    />
    <input
      className="input--block input--password mt-input"
      name="password"
      onChange={[Function]}
      placeholder="Contraseña"
      type="password"
      value=""
    />
  </div>
  <a
    className="mt-third pointer"
    href="#"
  >
    ¿Olvidaste tu contraseña?
  </a>
  <button
    className="btn__primary btn-block mt-secondary"
    disabled={false}
    onClick={[Function]}
  >
    Iniciar sesión
  </button>
</div>
`;
